test/utils.ts

```diff
@@ -47,13 +53,17 @@ export function wait(ms: number) {
 }
 
 // Assert that the given span's duration is within the given range.
-export function assertSpanDuration(span: TraceSpan, bounds: [number, number]) {
+export function assertSpanDuration(span: TraceSpan, bounds: [number, number?]) {
   const spanDuration = Date.parse(span.endTime) - Date.parse(span.startTime);
+  const lowerBound = bounds[0];
+  const upperBound = bounds[1] !== undefined ? bounds[1] : bounds[0];
   assert.ok(
-      spanDuration >= bounds[0] && spanDuration <= bounds[1],
+      spanDuration >= lowerBound - ASSERT_SPAN_TIME_TOLERANCE_MS &&
+          spanDuration <= upperBound + ASSERT_SPAN_TIME_TOLERANCE_MS,
```

Comment:
Do we need leniency on both sides of the test here? I would expect that the upper bound would be rigid.

Reply:
I added it for simplicity's sake. I don't see any opportunity for false positives as long as we keep the interval small.